src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X500DistinguishedNameBuilder.cs

```diff
@@ -0,0 +1,479 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Formats.Asn1;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    /// <summary>
+    /// This class facilitates building a distinguished name for an X.509 certificate.
+    /// </summary>
+    public sealed class X500DistinguishedNameBuilder
```

Comment:
If you can come up with a sane way to describe the order of operations here, that'd probably make for a good remarks section.

Assuming that we went with the user-sanity implementation over the code-sanity implementation, that'd be finding some way to explain something like

> For an X500DistinguishedName value whose Name property prints as "CN=widgets.example.org, OU=Widget Processing Center, O=Example Org" you would call AddCommonName("widgets.example.org"), then AddOrganizationalUnit("Widget Processing Center"), then AddOrganization("Example Org"), and finally Encode().

Or maybe it's only confusing when you're over-burdened with knowledge.  That this reverses, and X500DN reverses, so everything just works like anyone who doesn't know all this actually works thinks it does.

Reply:
I came up with a way, though I'll let you be the judge of it's sanity.